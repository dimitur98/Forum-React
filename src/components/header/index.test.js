import React from 'react'
import renderer from 'react-test-renderer'
import Header from '.'
import TestingEnvironment from '../../test-utils/router'

describe('Header', ()=>{
    it('should render routes for logged in user', () =>{
        const tree = renderer.create(
            <TestingEnvironment value={{
                user:{
                    email: "test"
                },
                loggedIn: true
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('should render routes for not logged in user', () =>{
        const tree = renderer.create(
            <TestingEnvironment value={{
                user:null,
                loggedIn: false
            }}>
                <Header/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})