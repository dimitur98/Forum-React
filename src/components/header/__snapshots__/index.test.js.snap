// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render routes for logged in user 1`] = `
<header>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-white border-bottom box-shadow mb-3"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
      >
        <img
          className="logo"
          src="logoDForum.jpg"
        />
      </a>
      <div
        className="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-dark"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-dark"
              href="/Account"
              onClick={[Function]}
            >
              Hi 
              test
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-dark"
              href=""
              onClick={[Function]}
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header should render routes for not logged in user 1`] = `
<header>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-white border-bottom box-shadow mb-3"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
      >
        <img
          className="logo"
          src="logoDForum.jpg"
        />
      </a>
      <div
        className="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link text-dark"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              class="nav-link text-dark"
              href="/Login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              class="nav-link text-dark"
              href="/Register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;
